### DISCUSSION TITLE
Standalone Model outside this Repository

### DISCUSSION THREAD

**Discussion by shenoynikhil** [ACEsuit/mace#119] [Q&A]
**Is your feature request related to a problem? Please describe.**
I want to use this model to train on energy and forces on the Spice dataset. Instead of converting my data into the xyz format and using it here, I want to incorporate it in my codebase in a way that allows me to load it and use it on any dataset. One of the reasons is also benchmarking it against other networks implemented in my repository. Is there a way to do this?

**Describe the solution you'd like**
I would like something like,
```python
from mace import MACE
```

I know that the above happens here as well, but I see a lot of stuff happening in the train script which makes it convoluted.

**Comment by ilyes319** [comment]
Hey @shenoynikhil, thank you for email.

This is entirely possible. Obviously you would need some of the arguments of run_train to create the model. You can look at the model test to check how it is done: [test_models.py](https://github.com/ACEsuit/mace/blob/develop/tests/test_models.py)

**Comment by shenoynikhil** [comment]
This looks great! Thank you so much. I'm closing the issue.

**Comment by shenoynikhil** [comment]
I was trying to use AtomicData object with pytorch_geometric DataLoader. Because its not `torch_geometric.data.Data`, the `Batch.from_data_list([<list of AtomicData>])` does not work. Do you have an idea of what could be done to fix this? 

I saw that you guys have copied the dataloader, collate stuff in `mace.tools.torch_geometric` and use that for dataloader but I could not figure out the difference that allows this to be loaded.

**Comment by davkovacs** [comment]
Could you give a little more detail of the problem? I recommend that you import the data loader from mace, not from torch geometric.

**Comment by shenoynikhil** [comment]
Currently, if I just install pytorch_geometric and use their `Batch.from_data_list([<list of AtomicData>])` it does not run. I was able to get it to run by copying your `batch.py`.

I do not want to use your torch_geometric since I am benchmarking this with other networks and other molecular graph datasets and want to use the latest `pyg`.

**Comment by ilyes319** [comment]
We use a lightweight version of Pytorch geometric to ease the maintenance and installation. Pytorch geometric is notoriously hard to install due to Cuda extensions. The mace models now take as input a dictionary of tensors. To make your code compatible, create the dictionaries with the correct entries. You should be able to do that from any initial Python object.

**Comment by shenoynikhil** [comment]
Okay, is there any documentation (`positions`, `edge_index`, `node_attrs` etc) on what keys would be required, or is that something for which I would need to go through the code?

**Comment by ilyes319** [comment]
Currently, it is not documented, but very easy to check by looking at the same test [test_models.py](https://github.com/ACEsuit/mace/blob/develop/tests/test_models.py) and printing the dictionary.

**Comment by ilyes319** [comment]
I want to add that training mace with the provided training script is highly encouraged, as many optimization details are crucial to mace performance.

**Comment by shenoynikhil** [comment]
The thing is, I want to test training this model on QM datasets that only have energies (also not present in xyz format). And I believe if the model is not tied to your repo and data utils, it might be easier to adopt. Kind of like how dimenet++ is present in pyg.

**Comment by ilyes319** [comment]
Only energies are also supported currently. I just wanted to warn you that training these ML force fields can be tricky, and I recommend you at least train on this repo with the default optimisation procedure to have a reference performance to compare to. 

I have made another repo [mace-layer](https://github.com/ACEsuit/mace-layer/tree/main) that lets your import a mace layer and stack it however you like with very standard and documented inputs also.

**Comment by shenoynikhil** [comment]
>Only energies are also supported currently. I just wanted to warn you that training these ML force fields can be tricky, and I recommend you at least train on this repo with the default optimisation procedure to have a reference performance to compare to

Good Option. I'll make sure to do that.

> I have made another repo [mace-layer](https://github.com/ACEsuit/mace-layer/tree/main) that lets your import a mace layer and stack it however you like with very standard and documented inputs also.

Great, will check it out.
